Update the notebook to have training code

annainspace opened the pull request:
Update so the notebook is able to get resources that have been pretrained as well as being robust against customers accidentally re-running any cell

*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

annainspace committed:
Update the notebook to have training code

Update so the notebook is able to get resources that have been pretrained as well as being
robust against customers accidentally re-running any cell

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
```diff
@@ -17,49 +17,25 @@
     "### In this notebook we will accomplish the following:\n",
     "\n",
-    "Create Video on Demand Domain Recommenders for the following use cases:\n",
+    "Walk you throuw how we created Video on Demand Domain Recommenders for the following use cases:\n",
```
throuw -> through

annainspace replied:
Fixing!

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
```diff
@@ -17,49 +17,25 @@
     "1. [Top picks for you](https://docs.aws.amazon.com/personalize/latest/dg/VIDEO_ON_DEMAND-use-cases.html#top-picks-use-case): personalized content recommendations for a user that you specify. With this use case, Amazon Personalize automatically filters videos the user watched based on the userId that you specify and Watch events.\n",
     "\n",
-    "Create a custom solution and solution versions for the following use case:\n",
+    "Walk you throuw how we created custom solution and solution versions for the following use case:\n",
```
throuw -> through

annainspace replied:
Fixing!

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
```diff
@@ -111,9 +87,1051 @@
+    "\n",
+    "As mentioned at the top of this notebook, a dataset group, schemas, datasets, solutions, and campaigns have already been created for you. You can open another browser tab/window to view these resources in the Personalize AWS Console.\n",
+    "\n",
+    "Bellow we will walk you though the steps we used to create there resorces, since they are already created, we will only be retrieving the automated deployment variables, however you can also run this code to train reources if you have not run the automation.\n",
```
though -> through
Bellow -> Below
there -> these or the
resorces -> resources

I think splitting this sentence up reads a little clearer:

_Below, we will walk you though the steps we used to create these resources. Since they are already created, we will only be retrieving the automated deployment variables, however, you can also run this code to train resources if you have not run the automation._

annainspace replied:
Fixing!

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
```diff
@@ -111,9 +87,1051 @@
+    "\n",
+    "Bellow we will walk you though the steps we used to create there resorces, since they are already created, we will only be retrieving the automated deployment variables, however you can also run this code to train reources if you have not run the automation.\n",
+    "\n",
+    "Please take into account that creating these resources in your account will incur a cost and that it will take upload and training time to to these steps trough this notebook (several hours)."
```
Corrected some spelling and missing words:

Please take into account that creating these resources in your account will incur a cost and that it will take upload and training time to complete these steps through this notebook (several hours).

annainspace replied:
Fixing!

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
```diff
@@ -111,9 +87,1051 @@
+    "\n",
+    "So far, we have downloaded, manipulated, and saved the data onto the Amazon EBS instance attached to instance running this Jupyter notebook. \n",
+    "\n",
+    "By default, the Personalize service does not have permission to acccess the data we uploaded into the S3 bucket in our account. In order to grant access to the  Personalize service to read our CSVs, we need to set a Bucket Policy and create an IAM role that the Amazon Personalize service will assume. Let's set all of that up.\n",
```
Spacing and spelling:

_By default, the Personalize service does not have permission to access the data we uploaded into the S3 bucket in our account. In order to grant access to the Personalize service to read our CSVs, we need to set a Bucket Policy and create an IAM role that the Amazon Personalize service will assume. Let's set all of that up._

annainspace replied:
Fixing!

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
This doesn't read quite right.  Reword to be more clear and not use the overloaded term "items".

_Before we can use the Dataset Group to create more resources below, it must be active. This can take a minute or two. Execute the cell below and wait for it to show the ACTIVE status. It checks the status of the dataset group every 60 seconds, up to a maximum of 3 hours._

annainspace replied:
Fixing!

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
Spacing and commas

Now that we've loaded and prepared our three datasets we'll configure the Amazon Personalize service to understand our data so that it can be used to train models for generating recommendations. Amazon Personalize requires a schema for each dataset, so it can map the columns in our CSVs to fields for model training. Each schema is declared in JSON using the [Apache Avro](https://avro.apache.org/) format.

annainspace replied:
Fixing!

aburkleaux-amazon reviewed (commented):
I know it is not part of the PR, but all the TOC links seem to be broken in this file.  They take me back to the README for the git repo.

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
```diff
@@ -111,9 +87,1051 @@
    "metadata": {},
    "source": [
-    "## Retrieve your automated deployment variables\n",
+    "# How to train your Use Case Optimized Recommenders and Solution Versions\n",
```
Use Case Optimized Recommenders is not a proper name.  Suggest:

"use case-optimized recommenders" OR "use-case-optimized recommenders"

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
```diff
@@ -111,9 +87,1051 @@
+    "\n",
+    "Use the metadata stored on the instance underlying this Amazon SageMaker notebook, to determine the region it is operating in. If you are using a Jupyter notebook outside of Amazon SageMaker, simply define the region as a string below. The Amazon S3 bucket needs to be in the same region as the Amazon Personalize resources we have been creating so far.\n",
+    "\n",
+    "First, let us get the current notebook region. I you are running your Amazon personalize experiments in a different region than your notebooks, you can specify it here."
```
typo: If you ...

annainspace replied:
Fixing!

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
Use consistent capitalization in headings.  Either all cap except connector words or sentence capitalization.  I prefer the latter.

_Wait for Dataset Group to have ACTIVE status_

annainspace replied:
Fixing!

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
Should this be: 

"There are several mandatory fields that are required in the schema, depending on the type of dataset"

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
until

annainspace replied:
Fixing!

aburkleaux-amazon commented on notebooks/Media-Pretrained/02_Training_Layer_Recap.ipynb:
```diff
@@ -17,49 +17,25 @@
-    "![Workflow](images/image2.png)\n",
-    "\n",
-    "To run this notebook, you need to have run the previous notebook, `01_Data_Layer.ipynb`, where you created a dataset and imported interaction, item, and user metadata data into Amazon Personalize. At the end of that notebook, you saved some of the variable values, which you now need to load into this notebook."
+    "![Workflow](images/image2.png)\n"
```
It would be helpful to introduce or caption this diagram.  Something like:

The following diagram shows the resources that we will create in this section, highlighted in blue.

annainspace replied:
Fixing!

aburkleaux-amazon reviewed (changes requested):
Please see my inline comments.

annainspace committed:
Fix spelling and wording

aburkleaux-amazon reviewed (approved):
Looks good!

annainspace committed:
Update the training notebook and merge with optional training

Added the .json parameter file and pull naming conventions from there so the notebook matches
the automation. This will make it easier to maintain.
The notebook looks for existing resources and only cerates them if they don't exist. Naming
paths are based on Service srn naming conventions.
Remove the old Optional training notebook and rename the forllowing notebooks accordingly.

annainspace committed:
Fix unnesserary print and markdon typo

aburkleaux-amazon reviewed (approved):
I looked at the latest changes and they look good.

annainspace committed:
Fix unnessary prints and fixed misleading text

annainspace committed:
Fix missing saved values

annainspace committed:
Fix issues with getting metrics and index link
